cmd/cli/main.go

```diff
@@ -0,0 +1,375 @@
+	}
+
+	s3 = v1.RestoreMethod{
+		S3: &v1.S3Spec{
+			Endpoint: restore.Cfg.RestoreToS3Endpoint,
+			Bucket:   restore.Cfg.RestoreToS3Bucket,
+			AccessKeyIDSecretRef: &corev1.SecretKeySelector{
+				Key: restore.Cfg.RestoreToS3SecretUsernameKey,
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: restore.Cfg.RestoreToS3Secret,
+				},
+			},
+			SecretAccessKeySecretRef: &corev1.SecretKeySelector{
+				Key: restore.Cfg.RestoreToS3SecretPasswordKey,
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: restore.Cfg.RestoreToS3Secret,
+				},
+			},
+		},
+	}
+
+	pvc = v1.RestoreMethod{
+		Folder: &v1.FolderRestore{
+			PersistentVolumeClaimVolumeSource: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: restore.Cfg.ClaimName,
+			},
+		},
+	}
+
+	restoreObject := v1.Restore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      restoreName,
+			Namespace: restore.Cfg.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Restore",
+			APIVersion: "k8up.io/v1",
+		},
+		Spec: v1.RestoreSpec{
+			//Snapshot: snapshot,
+			RunnableSpec: v1.RunnableSpec{
+				PodSecurityContext: &corev1.PodSecurityContext{
+					RunAsUser: &restore.Cfg.RunAsUser,
+				},
+				Backend: &v1.Backend{
+					RepoPasswordSecretRef: &corev1.SecretKeySelector{
+						Key: restore.Cfg.SecretRefKey,
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: restore.Cfg.SecretRef,
+						},
+					},
+					S3: &v1.S3Spec{
+						Endpoint: restore.Cfg.S3Endpoint,
+						Bucket:   restore.Cfg.S3Bucket,
+						AccessKeyIDSecretRef: &corev1.SecretKeySelector{
+							Key: restore.Cfg.S3SecretRefUsernameKey,
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: restore.Cfg.S3SecretRef,
+							},
+						},
+						SecretAccessKeySecretRef: &corev1.SecretKeySelector{
+							Key: restore.Cfg.S3SecretRefPasswordKey,
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: restore.Cfg.S3SecretRef,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	if restore.Cfg.RestoreMethod == "s3" {
+		restoreObject.Spec.RestoreMethod = &s3
+	} else {
+		restoreObject.Spec.RestoreMethod = &pvc
+	}
+
+	yamled, err := json.Marshal(restoreObject)
+	if err != nil {
+		logger.Error(err, "Failed to marshal restoreObject")
+		return err
+	}
+	post := client.RESTClient().Post().Namespace("default").AbsPath("/apis/k8up.io/v1/").Resource("restores").Body(yamled).Do(context.TODO())
+
+	status, err := post.Raw()
+	if err != nil {
+		// lines below looks very weird, but it's necessary, as actual reason why request failed is hidden in status.Message variable
+		var out1 metav1.Status
+		err := json.Unmarshal(status, &out1)
+		if err != nil {
+			logger.Error(err, "Failed to unmarshal Status object")
+		}
+		logger.Error(err, out1.Message)
+		return err
+	}
+	logger.Info(fmt.Sprintf("Backup created successfully, You can find it running:\tkubectl -n %s get restores.k8up.io %s", restore.Cfg.Namespace, restoreName))
```

Comment:
I would avoid adding suggestions in the log. We better add them in documentation or in the description when running the app with `--help` argument.

Reply:
why? I really like when software do something like that :D

Reply:
I don't see a problem here either.

Reply:
All these kind of suggestions on how restore works or what are the next steps after the restore has been done is very much useful in the docs, either in the repo or in the cli itself. Logs are supposed to tell what steps have been done by the application not what the user is supposed to do. Duplicating this log information for each restore does not seem very wise though I can leave with it as long as this is also documented in cli help or in the repo.

Reply:
I'll add it to docs additionally before sprint review

Reply:
If the docs will be updated then I'm fine with it.